### PR TITLE
Fixes #33537 -  Filter on packages/module streams with no errata

### PR DESCRIPTION
**Pull request by parthaa**
This commit adds UI bindings to Include / Exclude module streams and
packages with no errata.

### CODE REVIEW

**Comment by parthaa** [comment]
Content-View-Details (1)
Content-View-Details

File: webpack/scenes/ContentViews/Details/Filters/CVModuleStreamFilterContent.js [hunk 1]
```diff
@@ -209,6 +216,15 @@ const CVModuleStreamFilterContent = ({
                   }
                   />
                 </SplitItem>
+                <SplitItem>
+                  <Checkbox
+                    id="moduleStreamNoErrata"
+                    name="moduleStreamNoErrata"
+                    label={originalStreamsLabel}
```
Comment (sjha4) [inline webpack/scenes/ContentViews/Details/Filters/CVModuleStreamFilterContent.js:223]:
```suggestion
                   label=<p style={{marginTop:'4px'}}>{originalStreamsLabel}</p>
```

The checkboxes have a 4px margin added which doesn't apply to labels. This makes it better aligned on the page.

File: webpack/scenes/ContentViews/Details/Filters/CVModuleStreamFilterContent.js [hunk 2]
```diff
@@ -100,6 +101,11 @@ const CVModuleStreamFilterContent = ({
     });
   }, [filterResults, filterId]);
 
+  const enableOriginalModuleStreams = (checked) => {
+    setOriginalModuleStreams(checked);
+    dispatch(editCVFilter(filterId, { original_module_streams: checked }));
```
Comment (sjha4) [inline webpack/scenes/ContentViews/Details/Filters/CVModuleStreamFilterContent.js:106]:
```suggestion
    setOriginalModueStreamsUpdating(true);
    setOriginalModuleStreams(checked);
    dispatch(editCVFilter(filterId, { original_module_streams: checked }, () => {setOriginalModueStreamsUpdating(false)}));
```

Something like this with a state to disable checkbox when updating would prevent multiple clicks..

**Review by sjha4 — COMMENTED** [review]
Works great..Some nitpicks..

File: webpack/scenes/ContentViews/Details/Filters/CVModuleStreamFilterContent.js (hunk 2 above)
Comment (parthaa) [reply webpack/scenes/ContentViews/Details/Filters/CVModuleStreamFilterContent.js]: updated!.

File: webpack/scenes/ContentViews/Details/Filters/CVModuleStreamFilterContent.js (hunk 1 above)
Comment (parthaa) [reply webpack/scenes/ContentViews/Details/Filters/CVModuleStreamFilterContent.js]: updated!.

**Review by sjha4 — APPROVED** [review]
Works great! Thanks @parthaa ! 🥳

File: webpack/scenes/ContentViews/Details/Filters/ArtifactsWithNoErrata.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      useState(filterDetails[artifactAttribute] === true);
+
+  const [isLoading, setLoading] = useState(false);
+
+  const setArtifactsNoErrata = (checked) => {
+    enableArtifactsNoErrata(checked);
+    setLoading(true);
+    dispatch(editCVFilter(id, { [artifactAttribute]: checked }, () => setLoading(false)));
+  };
+
```
Comment (Andrewgdewar) [inline webpack/scenes/ContentViews/Details/Filters/ArtifactsWithNoErrata.js:23]: switch (true) {
case type ==='modulemd' && inclusion: return blah 
case type ==='modulemd' && !inclusion: return blah 
case  inclusion: return blah 
default return blah
} 

The above is a pattern I often use to clean up this kind of logic. But it's very much an optional change if you like it.

**Pull request merged**